pysat/tests/instrument_test_class.py

```diff
@@ -117,6 +117,13 @@ def generate_instrument_list(package=None):
 class InstTestClass():
     """Provides standardized tests for pysat instrument libraries.
     """
+    module_attrs = ['platform', 'name', 'tags', 'sat_ids',
+                    'load', 'list_files', 'download']
+    inst_attrs = ['tag', 'sat_id', 'acknowledgements', 'references']
+    inst_callable = ['load', 'list_files', 'download', 'clean', 'default']
+    attr_types = {'platform': str, 'name': str, 'tags': dict,
+                  'sat_ids': dict, 'tag': str, 'sat_id': str,
+                  'acknowledgements': str, 'references': str}
```

Comment:
Should there be a warning if _test_dates is not found?

Do we need a test for optional parameters that can be set at the instrument module level? 
Thinking it may be possible a person triggers an optional attribute without knowing.
Alternately, a pysat* package may want to test that optional attributes are present and/or valid.

I checked the code and here are the one pysat looks for.
        directory_format - updates instantiation default used for that particular instrument
        file_format - updates instantiation default used for that particular instrument
        multi_file_day - updates instantiation default used for that particular instrument
        orbit_info - provides instantiation default for that particular instrument
        pandas_format - pandas flag

There is a lapse in documentation on this as well (check against v2.2.1 online)

Reply:
There's another test in this class for test dates. I think optional parameters are out of scope here, since this is the required changes needed to migrate code out.

Reply:
Docstrings are handled in #487 and the upcoming mystery pull. 😺

Updates to the docs are a separate pull (not yet started). Just added a bit for the new features, but they will ultimately need a major update.

Reply:
> There's another test in this class for test dates. I think optional parameters are out of scope here, since this is the required changes needed to migrate code out.

Thanks.

> Updates to the docs are a separate pull (not yet started).

I miscommunicated. Not in the updated docs either. Fair enough to call it out of scope. Creating issue so we don't lose track.

Reply:
issue #512 created